cohortextractor/tpp_backend.py

```diff
@@ -653,6 +653,9 @@ def get_date_condition(self, table, date_expr, between):
         ]
         join_str = "\n".join(joins)
         if min_date_expr is not None and max_date_expr is not None:
+            if flags.WITH_END_DATE_FIX:
+                min_date_expr = MSSQLDateFormatter.cast_as_date(min_date_expr)
+                max_date_expr = MSSQLDateFormatter.cast_as_date(max_date_expr)
```

Comment:
Drive-by comment: do we also need to make this change on lines 664 and 666?

Reply:
quite possibly - we've only specifically investigated the behaviour wrt. `BETWEEN` but it stands to reason (and in my testing on a dummy table with mssql) that we should change those also